### PR TITLE
[se-1040] K8s review

### PR DESCRIPTION
**Pull request by ritwika314**
Companion PR for https://github.com/Bodo-inc/Bodo/pull/7529

### CODE REVIEW

File: 08-Kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+## Prerequisites
+
+- Access to a Kubernetes cluster such as AWS EKS. To start a new Kubernetes Cluster in EKS, you can follow our guide here: [Create EKS Cluster](#create-eks-cluster-using-kops-optional).
+
+- Create a docker image that contains your intended Bodo version and Python scripts and upload it to a docker registry such as Docker Hub so that Kubernetes can pull it. 
+For reference, see this [Dockerfile](Dockerfile).
+- We have included a python script called `pi.py` which can be used for validating your setup.
+
+A docker image created from this Dockerfile is also available on DockerHub: [ritwikabodo/bodo-kubernetes]().
```
Comment (ehsantn) [inline 08-Kubernetes/README.md:13]: This should be changed to bodoai Docker Hub.

File: 08-Kubernetes/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+mpijobs.kubeflow.org    2024-04-02T19:43:04Z
+```
+
+#### Step 2: Run your Bodo application
+
+- Define a kubernetes resource for your Bodo workload, such as the one defined in [`mpijob.yaml`](mpijob.yaml) that runs the [pi example](pi.py). You can modify it based on your cluster configuration: 
+
+1. Update `spec.slotsPerWorker` with the number of physical cores (_not_ vCPUs) on each node
+2. Set `spec.mpiReplicaSpecs.Worker.replicas` to the number of worker nodes in your cluster.
+3. Build the image using the Dockerfile or use `ritwikabodo/bodo-kubernetes` and replace the image at
```
Comment (ehsantn) [inline 08-Kubernetes/README.md:56]: Again, use bodoai registery.

File: 08-Kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+- Attempt to create your cluster: 
+
+This creates a cluster of 2 nodes each with 4 cores. To change the number of instances, modify the `node-count` argument and to change the worker nodes update `node-size`. `master-size` refers to the leader that manages K8s but doesn’t do any Bodo computation, so you should keep the instance small. You can deploy the cluster in a different AWS region and availability zone by modifying the `zones` argument. 
+
+```
+kops create cluster \
+--node-count=2 \
+--node-size=c5.2xlarge \
+--master-size=c5.large \
```
Comment (ehsantn) [inline 08-Kubernetes/README.md:117]: Didn't you change this to `control-plane` in this PR? https://github.com/Bodo-inc/Bodo/pull/7529/files

**Review by ehsantn — APPROVED** [review]
Looks good to me, but would be great for someone familiar with k8s to try it out and give feedback as well.

File: 08-Kubernetes/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM mpioperator/intel:latest
+
+COPY --from=build /opt/conda /opt/conda
+
+RUN chown -R mpiuser /opt/conda
+ENV PATH /opt/conda/bin:${PATH}
+USER mpiuser
+
+ADD pi.py pi.py
```
Comment (bodo-sai) [inline 08-Kubernetes/Dockerfile:13]: Can you take a look at this folder where I have created 2 Dockerfiles for Apple? One dockerfile which is similar one to this but with less size is called `Dockerfile` and another one with EFA enabled which is built from scratch called `Dockerfile-efa`

https://github.com/Bodo-inc/customer-sample-code/tree/master/apple/insight/eks-setup/docker

File: 08-Kubernetes/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
+FROM condaforge/mambaforge as build
+
+RUN conda install -y python=3.12 bodo=2024.3 mpich=4.1.2 -c bodo.ai -c conda-forge
+
+FROM mpioperator/intel:latest
```
Comment (IsaacWarren) [inline 08-Kubernetes/Dockerfile:5]:
```suggestion
FROM mpioperator/intel:0.4.0
```

**Pull request merged**